docs/specs/apb.yml-plan-support.md

```diff
@@ -0,0 +1,165 @@
+    default: mediawiki
+    type: string
+    title: Mediawiki DB Schema
+  - name: mediawiki_site_name
+    default: MediaWiki
+    type: string
+    title: Mediawiki Site Name
+```
+
+Should use this pattern for any additional changes moving forward.
+
+---
+
+4) Remove `required` section. `required` field should return to being an attribute on `Parameter`s.
+
+```yaml
+parameters:
+  - name: mediawiki_db_schema
+    required: true
+    default: mediawiki
+    type: string
+    title: Mediawiki DB Schema
+```
+
+## Notes
+
+* Use yaml anchors and inheritence to cut down on parameter duplication if plans share
+parameters, see full example.
+
+## Questions
+
+* Obviously this is a large, breaking change to **all** existing APBs and their `apb.yml` files.
+How do we want to make across the board so things get rolled out as smoothly as possible?
+* Is this a good opportunity to have the broker start respecting the version label found on
+apbs? I think it makes sense to have that in lock setup with broker release versions. Is
+broker master considered `0.10` since it's `release-0.9++`?
+
+## Full example
+
+```yaml
+################################################################################
+# Shared Parameters
+################################################################################
+_p: &_p
+  - name: postgresql_database
+    default: admin
+    type: string
+    title: PostgreSQL Database Name
+    required: true
+  - name: postgresql_password
+    type: string
+    description: A random alphanumeric string if left blank
+    title: PostgreSQL Password
+  - name: postgresql_user
+    default: admin
+    title: PostgreSQL User
+    type: string
+    maxlength: 63
+    required: true
+  - name: postgresql_version
+    default: 9.5
+    enum: ['9.5', '9.4']
+    type: enum
+    title: PostgreSQL Version
+    required: true
+################################################################################
+name: rhscl-postgresql-apb
+image: eriknelson/rhscl-postgresql-apb
+description: SCL PostgreSQL apb implementation
+bindable: True
+async: optional
+metadata:
+  documentationUrl: https://www.postgresql.org/docs/
+  longDescription: An apb that deploys postgresql 9.4 or 9.5.
+  dependencies: ['registry.access.redhat.com/rhscl/postgresql-95-rhel7']
+  displayName: PostgreSQL (APB)
+  console.openshift.io/iconClass: icon-postgresql
+dependencies:
+  - postgresql_version
+plans:
+  - name: dev
+    description: A single DB server with no storage
+    free: true
+    metadata:
+      displayName: Development
+      longDescription: This plan provides a single non-HA PostgreSQL server without persistent storage
+      cost: $0.00
+    parameters: *_p
+  - name: prod
+    description: HA DB Server with 1TB of Storage
+    free: false
+    metadata:
+      displayName: Production
+      longDescription: This plan provides a single non-HA PostgreSQL server with persistent storage
+      cost: $5.99 monthly
+    parameters: *_p
+```
```

Comment:
Would be nice to see a simple example of not using the references so folks know you can also put the parameters inline with the plan.